Userland/Libraries/LibGL/GLContext.h

```diff
@@ -271,6 +272,13 @@ class GLContext final {
     GLenum m_current_read_buffer = GL_BACK;
     GLenum m_current_draw_buffer = GL_BACK;
 
+    // User-defined clip planes
+    struct ClipPlaneAttributes {
+        FloatVector4 eye_clip_plane[6]; // TODO: Change to use device-defined constant
+        FloatVector4 proj_clip_plane[6];
```

Comment:
If you leave it as arrays, you can also use `Array<FloatVector4, 6>` instead. Otherwise, `Vector<FloatVector4>` might also be fine in order to support an arbitrary amount of user clip planes. Let `Device` determine the limit.

Reply:
I chose to keep it as a 6 element array for the time being, just so it plays nicely with the current enable/disable clip plane code.